tests/unit/src/test/scala/tests/WindowStateDidChangeSlowSuite.scala

```diff
@@ -0,0 +1,42 @@
+package tests
+
+object WindowStateDidChangeSlowSuite
+    extends BaseSlowSuite("window-state-did-change") {
+  testAsync("basic") {
+    for {
+      _ <- server.initialize(
+        """
+          |/metals.json
+          |{ "a": {} }
+          |/a/src/main/scala/a/A.scala
+          |package a
+          |object A {
+          |  val x = 42
+          |}
+        """.stripMargin
+      )
+      _ <- server.didOpen("a/src/main/scala/a/A.scala")
+      _ = assertNoDiagnostics()
+
+      _ = server.windowStateDidChange(focused = false)
+      _ = assert(server.server.pauseables.isPaused.get())
+      didChange = server.didChange("a/src/main/scala/a/A.scala")(
```

Comment:
I refactored this test to be based on `didChange` and syntax errors in order to reduce the number of sleep milliseconds. I tested manually that removing the `server.windowStateDidChange(focused = false)` statement makes the test fail.